packages/build/src/local/trigger-release-draft.ts

```diff
@@ -0,0 +1,98 @@
+import assert from 'assert';
+import semver from 'semver';
+import { choose, confirm, spawnSync as spawnSyncFn } from '../helpers';
+import { getLatestDraftOrReleaseTagFromLog, TagDetails } from './get-latest-tag';
+import { getRepositoryStatus } from './repository-status';
+
+type BumpType = 'draft' | 'patch' | 'minor' | 'major';
+
+export async function triggerReleaseDraft(
+  repositoryRoot: string,
+  spawnSync: typeof spawnSyncFn = spawnSyncFn
+): Promise<void> {
+  console.info('Triggering process to create a new release draft...');
+
+  verifyGitStatus(repositoryRoot);
+
+  const latestDraftOrReleaseTag = getLatestDraftOrReleaseTagFromLog(repositoryRoot);
+  if (!latestDraftOrReleaseTag) {
+    throw new Error('Could not find a previous draft or release tag.');
+  }
+  console.info(`-> Most recent tag: v${latestDraftOrReleaseTag.tag.semverName} on commit ${latestDraftOrReleaseTag.commit}`);
+
+  let bumpType: BumpType = 'draft';
+  if (latestDraftOrReleaseTag.tag.draftVersion === undefined) {
+    bumpType = await choose('>  Select the type of increment for the new version', [
+      'patch', 'minor', 'major'
+    ], '... enter your choice:') as BumpType;
+  }
+
+  const nextTagName = computeNextTagName(latestDraftOrReleaseTag.tag, bumpType);
+  console.info('-> New draft tag is:');
+  console.info(`       ${nextTagName}`);
+
+  const confirmed = await confirm('!! Is this correct and should the draft process continue?');
+  if (!confirmed) {
+    throw new Error('User aborted.');
+  }
+
+  console.info('... creating and pushing tag ...');
+  spawnSync('git', ['tag', nextTagName], {
+    cwd: repositoryRoot,
+    encoding: 'utf-8'
+  });
+  spawnSync('git', ['push', '--tags'], {
+    cwd: repositoryRoot,
+    encoding: 'utf-8'
+  });
+
+  console.info('SUCCESS! Your new draft has been tagged and pushed.');
+}
+
+function verifyGitStatus(repositoryRoot: string): void {
+  const repositoryStatus = getRepositoryStatus(repositoryRoot);
+  if (!repositoryStatus.branch?.local) {
+    throw new Error('Could not determine local repository information - please verify your repository is intact.');
+  }
+
+  if (repositoryStatus.branch?.local !== 'master') {
+    throw new Error('Triggering a draft release is only allowed from master branch');
+  }
+  if (!repositoryStatus.branch.tracking) {
+    throw new Error('The branch you are on is not tracking any remote branch. Ensure you are on the master branch of the repository.');
+  }
```

Comment:
These things are actually moving us further away from being able to do support releases, right? Could we maybe prompt the user instead to make sure they’re on the right branch?

Reply:
Yeah it depends probably on how we want to do them. At least a tracking branch should be there. Right now I've put in the `master` guard just for good measure. I'd assume we create a separate branch for a support release and tag on that branch. But for that it should definitely be pushed to the remote (thus we should have a `tracking` branch).

Reply:
Yeah, I wasn’t referring to the tracking branch check specifically, sorry :) Maybe we could do some very crude regex check here, e.g. `/^master$|^main$|^v\d+\.\d+\.\d+/` (main because of https://jira.mongodb.org/browse/MONGOSH-245)?

Reply:
That's a good idea 👍 So we'd cover `master`/`main` and potential support release branches (which would equal to the designated version number)... ✅

Reply:
Hm yeah … and maybe also allow `[xyz0-9]` instead of `\d`, I don’t think we have any idea what the branch naming scheme will look like but using things like `v0.x.y` or `v0.8.x` would be common